python-package/packager/pep517.py

```diff
@@ -79,7 +79,8 @@ def build_wheel(
         libxgboost = locate_or_build_libxgboost(
             TOPLEVEL_DIR, build_dir=build_dir, build_config=build_config
         )
-        copy_with_logging(libxgboost, lib_path, logger=logger)
+        if not build_config.use_system_libxgboost:
+            copy_with_logging(libxgboost, lib_path, logger=logger)
```

Comment:
I manually tested this change by:
1. Create a new Conda environment.
2. Build XGBoost from the source using CMake:
```
conda activate my_env
cmake .. -GNinja -DCMAKE_INSTALL_PREFIX="$CONDA_PREFIX" -DCMAKE_INSTALL_LIBDIR="lib"
```
3. Install libxgboost into the Conda env:
```
ninja install
```
4. Install the Python package with `use_system_libxgboost` flag:
```
pip install -v .  --config-settings use_system_libxgboost=True
```
5. Inspect `$CONDA_PREFIX` to ensure that only one copy of `libxgboost.so` is installed in the Conda environment.